mne/decoding/transformer.py

```diff
@@ -12,42 +12,72 @@
 from .. import pick_types
 from ..filter import filter_data, _triage_filter_params
 from ..time_frequency.psd import _psd_multitaper
-from ..externals import six
-from ..utils import _check_type_picks
+from ..utils import _check_type_picks, requires_sklearn, check_version
+from ..io.pick import pick_info, _pick_data_channels, _picks_by_type
+from ..cov import _undo_scaling_array, _apply_scaling_array
 
 
-class Scaler(TransformerMixin):
-    """Standardize data across channels.
+@requires_sklearn
+class Scaler(TransformerMixin, BaseEstimator):
+    """Standardizes data across channels.
 
     Parameters
     ----------
+    scalings : dict, string, defaults to "standard"
+        if scalings is 'robust' sklearn.proprocessing.RobustScaler() is used.
+        robust requires sklearn version 0.17.
+        if scalings is 'standard' sklearn.preprocessing.StandardScaler()
+        is used.
+        if scalings is 'default', scales mag by 1e15
+                                         grad by 1e13
+                                         eeg by 1e6
+
     info : instance of Info
         The measurement info
     with_mean : boolean, True by default
-        If True, center the data before scaling.
+        If True, center the data using mean before scaling.
     with_std : boolean, True by default
         If True, scale the data to unit variance (or equivalently,
         unit standard deviation).
+    with_centering: boolean, True by default
+        If True, center the data using median before scaling.
+        This is only valid(used) in robust method.
 
     Attributes
     ----------
-    info : instance of Info
-        The measurement info
     ``ch_mean_`` : dict
         The mean value for each channel type
-    ``std_`` : dict
+    ``ch_median_`` : dict
+        The median value for each channel type
+    ``scale_`` : dict
         The standard deviation for each channel type
-    """
+     """
 
-    def __init__(self, info, with_mean=True, with_std=True):  # noqa: D102
+    def __init__(self, scalings='standard', info=None, with_mean=True,
+                 with_std=True, with_centering=True):
         self.info = info
         self.with_mean = with_mean
         self.with_std = with_std
-        self.ch_mean_ = dict()  # TODO rename attribute
-        self.std_ = dict()  # TODO rename attribute
+        self.with_centering = with_centering
+
+        if (isinstance(scalings, str) and scalings not in
+                      ('robust', 'standard', 'default')):
+            raise ValueError("Invalid method for scaling")
+
+        if not isinstance(scalings, (dict, str)):
+            raise ValueError("scalings type should be dict or str")
+
+        if scalings == 'robust':
+            if not check_version('sklearn', '0.17'):
```

Comment:
Doesn't look like its documented anywhere. I think a good place to add the minimum version(0.18) would be at `doc/manual/decoding.rst`. What do you think?